src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -1010,19 +1036,39 @@ AppendShardIdNameValues(StringInfo selectQuery, ShardInterval *shardInterval)
 
 
 /*
- * ErrorIfNotSuitableToGetSize determines whether the table is suitable to find
+ * ErrorIfNotSuitableToGetSize determines whether the relation is suitable to find
  * its' size with internal functions.
  */
 static void
 ErrorIfNotSuitableToGetSize(Oid relationId)
 {
 	if (!IsCitusTable(relationId))
 	{
-		char *relationName = get_rel_name(relationId);
-		char *escapedQueryString = quote_literal_cstr(relationName);
-		ereport(ERROR, (errcode(ERRCODE_INVALID_TABLE_DEFINITION),
-						errmsg("cannot calculate the size because relation %s is not "
-							   "distributed", escapedQueryString)));
+		if (get_rel_relkind(relationId) != RELKIND_INDEX)
+		{
+			char *relationName = get_rel_name(relationId);
+			char *escapedQueryString = quote_literal_cstr(relationName);
+			ereport(ERROR, (errcode(ERRCODE_INVALID_TABLE_DEFINITION),
+							errmsg(
+								"cannot calculate the size because relation %s "
+								"is not distributed",
+								escapedQueryString)));
+		}
+		bool missingOk = false;
+		Oid indexId = relationId;
+		relationId = IndexGetRelation(relationId, missingOk);
+		if (!IsCitusTable(relationId))
+		{
+			char *tableName = get_rel_name(relationId);
+			char *escapedTableName = quote_literal_cstr(tableName);
+			char *indexName = get_rel_name(indexId);
+			char *escapedIndexName = quote_literal_cstr(indexName);
+			ereport(ERROR, (errcode(ERRCODE_INVALID_OBJECT_DEFINITION),
+							errmsg(
+								"cannot calculate the size because table %s for "
+								"index %s is not distributed",
+								escapedTableName, escapedIndexName)));
+		}
```

Comment:
Some style improvements and variable renaming suggestions:

```suggestion
		if (get_rel_relkind(relationId) != RELKIND_INDEX)
		{
			char *relationName = get_rel_name(relationId);
			char *escapedRelationName = quote_literal_cstr(relationName);
			ereport(ERROR, (errcode(ERRCODE_INVALID_TABLE_DEFINITION),
							errmsg("cannot calculate the size because "
							       "relation %s is not distributed",
  								   escapedRelationName)));
		}

		bool missingOk = false;
		Oid indexId = relationId;
		relationId = IndexGetRelation(relationId, missingOk);
		if (!IsCitusTable(relationId))
		{
			char *tableName = get_rel_name(relationId);
			char *escapedTableName = quote_literal_cstr(tableName);
			char *indexName = get_rel_name(indexId);
			char *escapedIndexName = quote_literal_cstr(indexName);
			ereport(ERROR, (errcode(ERRCODE_INVALID_OBJECT_DEFINITION),
							errmsg("cannot calculate the size because "
							       "table %s for index %s is not "
							       "distributed",
								   escapedTableName, escapedIndexName)));
		}
```

Reply:
will do but the diff is too huge, probably needs citus_indent...

Reply:
added